lib/profile.js

```diff
@@ -86,52 +85,63 @@ const knownProfileKeys = [
   'updated',
 ]
 
-function get (args) {
+async function get (args) {
   const tfa = 'two-factor auth'
-  const conf = npm.flatOptions
-  return pulseTillDone.withPromise(profile.get(conf)).then((info) => {
-    if (!info.cidr_whitelist)
-      delete info.cidr_whitelist
-    if (conf.json) {
-      output(JSON.stringify(info, null, 2))
-      return
-    }
-    const cleaned = {}
-    knownProfileKeys.forEach((k) => {
-      cleaned[k] = info[k] || ''
-    })
-    Object.keys(info).filter((k) => !(k in cleaned)).forEach((k) => {
-      cleaned[k] = info[k] || ''
-    })
-    delete cleaned.tfa
-    delete cleaned.email_verified
-    cleaned.email += info.email_verified ? ' (verified)' : '(unverified)'
-    if (info.tfa && !info.tfa.pending)
-      cleaned[tfa] = info.tfa.mode
-    else
-      cleaned[tfa] = 'disabled'
-
-    if (args.length) {
-      const values = args // comma or space separated ↓
-        .join(',').split(/,/).map((arg) => arg.trim()).filter((arg) => arg !== '')
-        .map((arg) => cleaned[arg])
-        .join('\t')
-      output(values)
-    } else {
-      if (conf.parseable) {
-        Object.keys(info).forEach((key) => {
-          if (key === 'tfa')
-            output(`${key}\t${cleaned[tfa]}`)
-          else
-            output(`${key}\t${info[key]}`)
-        })
-      } else {
-        const table = new Table()
-        Object.keys(cleaned).forEach((k) => table.push({ [ansistyles.bright(k)]: cleaned[k] }))
-        output(table.toString())
+  const conf = { ...npm.flatOptions }
+
+  const info = await pulseTillDone.withPromise(npmProfile.get(conf))
+
+  if (!info.cidr_whitelist)
+    delete info.cidr_whitelist
+
+  if (conf.json) {
+    output(JSON.stringify(info, null, 2))
+    return
+  }
+
+  // clean up and format key/values for output
+  const cleaned = {}
+  for (const key of knownProfileKeys)
+    cleaned[key] = info[key] || ''
+
+  const unknownProfileKeys = Object.keys(info).filter((k) => !(k in cleaned))
+  for (const key of unknownProfileKeys)
+    cleaned[key] = info[key] || ''
+
+  delete cleaned.tfa
+  delete cleaned.email_verified
+  cleaned.email += info.email_verified ? ' (verified)' : '(unverified)'
+
+  if (info.tfa && !info.tfa.pending)
+    cleaned[tfa] = info.tfa.mode
+  else
+    cleaned[tfa] = 'disabled'
+
+  if (args.length) {
+    const values = args // comma or space separated
+      .join(',')
+      .split(/,/)
+      .map((arg) => arg.trim())
```

Comment:
this map can be removed, and the .trim done in the filter, since you replace the trimmed arg with cleaned[arg] anyways, and you’ll skip one of the intervening arrays.